### PR TITLE
Implement `imgui-sdl2-support` crate.

### PR DESCRIPTION
**Pull request by NightShade256**
This PR implements a SDL 2 support crate for `imgui-rs`, similar to the existing `imgui-winit-support` crate. This is mainly due to the fact that the existing SDL 2 support crate is severely outdated and an "official" crate is beneficial. This also checks off the last point from the list at the end of #444.

(sorry for the new PR!)

### CODE REVIEW

**Comment by NightShade256** [comment]
The only thing now left is adding top-level doc-comments and some other small doc changes. I am not sure what to do about locked/rounded DPI (also adding that I don't have much knowledge about the same), the existing third party SDL support crate also doesn't handle it.

**Review by sanbox-irl — CHANGES_REQUESTED** [review]
First off, thank you very much for this PR. This excellent, and exactly what we needed!

However, I'd like to ask that we make a few changes.

First, let's simplify button to no longer have `Cell` in it. We can do this within this code trivially, and then make `get` just take `&self` without mutating state, and then in a second loop reset all our buttons. I know this is slightly less elegant, but I think this isn't an appropriate `Cell` usage. I worry that in the future, maintainers would be less likely to touch that code since it seems "complicated". Personally, I also have redoing `Button` to just be an enum, like so:

```rs
#[derive(Debug, Clone, Copy)]
enum Button {
    /// This button was either released this frame, or is
    /// entirely unpressed.
    None,
    /// This button is down, with a bool indicating if it was
    /// pressed on this frame.
    Down(bool),
    /// This button is down but was released this frame too.
    PressedThisFrame,
}
```

But I leave that up to you!

Second:

Let's split up `handle_event` into `handle_event` and `filter_event`. Most users won't need to `filter_event`, but the optimizer won't be able to see that, so we're spinning some extra CPU cycles needlessly for most people. Plus, it'll let them filter using `.filter` and who doesn't love a good `.filter`. I left a bigger explanation in the review for this.

Third:

Let's combine `prepare_frame` and `prepare_render` into one function. When you wrote this, that wasn't possible, since we bound `ImGui::GetMouseCursor` behind the `Ui` object, but I have now pushed and added a `context.mouse_cursor`, so we should be able to get the request mouse_cursor ourselves.

This actually will mean that our mouse cursor requests will be delayed by one frame -- if you'd like to make an optional method that's like `prepare_mouse_cursor` with some doc comments indicating that it prevents mouse cursor lag, feel free. I think the one frame delay is fine, though, for two reasons: 1. users can set the mouse cursor AFTER called `prepare_render` but before `render`, and then that cursor change will actually never happen at all, and 2. the official sdl2 binding in dear imgui does exactly that. There's actually no reason we can't double set it, so adding a "you had frame delays desperately, use this method" is fine by me, though I wouldn't include it in examples.

That will also mean that `prepare_frame` would take `&mut Context` rather than `&mut Io`, and that actually seems correct to me -- we should get the `io` object ourselves, rather than make users give it to us.

What do you think of all of these changes? I know it's a lot, and I really appreciate the effort you're putting it. Sorry it's taken me a few days to write all this down; I just wanted to make sure to do a thorough review

File: imgui-sdl2-support/Cargo.toml
Comment (NightShade256) [reply imgui-sdl2-support/Cargo.toml:18]: The only two realistic choices were the glow renderer and the wgpu renderer, since the glow one was in-tree I went with that.

File: imgui-sdl2-support/LICENSE-MIT
Comment (NightShade256) [reply imgui-sdl2-support/LICENSE-MIT:1]: Sure, done!

File: imgui-sdl2-support/LICENSE-APACHE
Comment (NightShade256) [reply imgui-sdl2-support/LICENSE-APACHE:189]: I am not sure about that since that's merely an appendix section,  and all of the other crates in the tree leave it as is.

File: imgui-sdl2-support/examples/basic.rs
Comment (NightShade256) [reply imgui-sdl2-support/examples/basic.rs:60]: I was thinking about doing it too, but I backed off after seeing the in-tree `winit` support crate. But I think this is a great change so maybe we should port it over to the `winit` backend as well.

File: imgui-sdl2-support/src/lib.rs
Comment (NightShade256) [reply imgui-sdl2-support/src/lib.rs:171]: Yes, I usually separate `impl`s into "public" and "private" ones. About the concern, I feel that is a valid point. I think I will just move the private impl below the public one, or should I just merge them?

File: imgui-sdl2-support/src/lib.rs
Comment (NightShade256) [reply imgui-sdl2-support/src/lib.rs]: Just noticed that this doesn't need to be a method and can just be a normal function.

**Comment by NightShade256** [comment]
I am a bit busy the next day, so I will address the other points a bit later. Sorry!

File: imgui-sdl2-support/LICENSE-APACHE
Comment (sanbox-irl) [reply imgui-sdl2-support/LICENSE-APACHE:189]: hmm okay, let's leave it as is, and I'll do a copyright walk

File: imgui-sdl2-support/src/lib.rs
Comment (sanbox-irl) [reply imgui-sdl2-support/src/lib.rs]: Yeah let's move the private before the public one, and otherwise it's fine. I like developers to leave their own idiosyncrasies on their code, as long as they're not too strange. It's your work, after all

**Comment by sanbox-irl** [comment]
Good call -- when this is merged, I will make the same change to the winit crate!

And don't worry about timing -- I didn't review until Sunday night, so I understand that we've all got work/classes in the week

File: imgui-sdl2-support/src/lib.rs
Comment (NightShade256) [reply imgui-sdl2-support/src/lib.rs]: That is a great idea!

**Comment by sanbox-irl** [comment]
When the checks go green, I am going to:
1. rebase it (I don't think it actually needs it, but if it does, it'll have a force push, so your local might get a little messed up)
2. add a changelog entry crediting you.
3. merge it in!

Thank you so much!

**Pull request merged**
